File: api_service/src/services/auth/bootstrap_cli.py
from typing import cast

from src.models.auth.bootstrap import (
    BootstrapAuthRequest,
    BootstrapAuthResult,
    BootstrapStage,
    ChallengePayload,
    ChallengeRequest,
)
from src.services.auth.auth_gateway_authority import IAuthGatewayAuthorityClient


class BootstrapClient:
    """冷启动认证流程服务（只转发认证中心）。"""

    def __init__(self, authority_client: IAuthGatewayAuthorityClient):
        self._authority_client = authority_client

    async def init_challenge(self, req: ChallengeRequest) -> ChallengePayload:
        return await self._authority_client.init_challenge(req)

    async def authenticate_bootstrap(
        self, req: BootstrapAuthRequest
    ) -> BootstrapAuthResult:
        return await self._authority_client.authenticate_bootstrap(req)

    async def get_bootstrap_stage(self, ctx: object) -> BootstrapStage:
        principal_id = ""
        if isinstance(ctx, dict):
            principal_id = str(ctx.get("principal_id") or "")
        elif isinstance(ctx, str):
            principal_id = ctx
        principal_id = principal_id.strip()
        if not principal_id or ":" not in principal_id:
            return "uninitialized"
        entity_type, entity_id = principal_id.split(":", 1)
        entity_type = entity_type.strip()
        entity_id = entity_id.strip()
        if not entity_type or not entity_id:
            return "uninitialized"
        stage = await self._authority_client.get_bootstrap_stage(entity_type, entity_id)
        return cast(BootstrapStage, stage)

File: gateway/src/iface/auth/bootstrap_cli.go
package auth

import (
	"context"
	authmodel "gateway/src/models/auth"
)

// IBootstrapClient 定义外部调用认证中心冷启动认证流程的客户端接口，包含初始化挑战、验证挑战和查询冷启动阶段等方法。
type IBootstrapClient interface {
	// InitChallenge 向认证中心请求一次性挑战。
	InitChallenge(ctx context.Context, req *authmodel.ChallengeRequest) (*authmodel.ChallengePayload, error)
	// AuthenticateBootstrap 提交签名后的挑战并换取会话与令牌。
	AuthenticateBootstrap(ctx context.Context, req *authmodel.BootstrapAuthRequest) (*authmodel.BootstrapAuthResult, error)
	// GetBootstrapStage 返回当前模块冷启动状态，便于健康检查与启动门禁。
	GetBootstrapStage(ctx context.Context) (authmodel.BootstrapStage, error)
}
